cmd/appliance/upgrade/prepare.go

```diff
@@ -367,8 +408,98 @@ func prepareRun(cmd *cobra.Command, args []string, opts *prepareUpgradeOptions)
 		}
 	}
 
+	uploadWithProgress := func(ctx context.Context, reader io.Reader, name string, size int64, headers map[string]string) error {
+		p := tui.New(ctx, spinnerOut)
+		defer p.Wait()
+		endMsg := "uploaded"
+		proxyReader, t := p.FileUploadProgress(name, endMsg, size, reader)
```

Comment:
looks like you are missing `proxyReader.Close()`

Reply:
Yep, I'll fix that :+1: